src/code/code_80043480.c

```diff
@@ -3,73 +3,73 @@
 void DynaPolyActor_Init(DynaPolyActor* dynaActor, s32 flags) {
     dynaActor->bgId = -1;
     dynaActor->unk_15C = flags;
-    dynaActor->unk_160 = 0;
+    dynaActor->interactFlags = 0;
     dynaActor->unk_150 = 0.0f;
     dynaActor->unk_154 = 0.0f;
 }
 
-void func_800434A0(DynaPolyActor* dynaActor) {
-    dynaActor->unk_160 = 0;
+void DynaPolyActor_ResetInteractFlags(DynaPolyActor* dynaActor) {
+    dynaActor->interactFlags = 0;
 }
 
-void func_800434A8(DynaPolyActor* dynaActor) {
-    dynaActor->unk_160 |= 1;
+void DynaPolyActor_SetActorOnTop(DynaPolyActor* dynaActor) {
+    dynaActor->interactFlags |= DYNA_INTERACT_ACTOR_ON_TOP;
 }
 
-void func_800434B8(DynaPolyActor* dynaActor) {
-    dynaActor->unk_160 |= 2;
+void DynaPolyActor_SetPlayerOnTop(DynaPolyActor* dynaActor) {
+    dynaActor->interactFlags |= DYNA_INTERACT_PLAYER_ON_TOP;
 }
 
-void func_800434C8(CollisionContext* colCtx, s32 floorBgId) {
+void DynaPoly_SetPlayerOnTop(CollisionContext* colCtx, s32 floorBgId) {
     DynaPolyActor* dynaActor = DynaPoly_GetActor(colCtx, floorBgId);
 
     if (dynaActor != NULL) {
-        func_800434B8(dynaActor);
+        DynaPolyActor_SetPlayerOnTop(dynaActor);
     }
 }
 
-void func_800434F8(DynaPolyActor* dynaActor) {
-    dynaActor->unk_160 |= 4;
+void DynaPolyActor_SetPlayerAbove(DynaPolyActor* dynaActor) {
+    dynaActor->interactFlags |= DYNA_INTERACT_PLAYER_ABOVE;
 }
 
-void func_80043508(CollisionContext* colCtx, s32 floorBgId) {
+void DynaPoly_SetPlayerAbove(CollisionContext* colCtx, s32 floorBgId) {
     DynaPolyActor* dynaActor = DynaPoly_GetActor(colCtx, floorBgId);
 
     if (dynaActor != NULL) {
-        func_800434F8(dynaActor);
+        DynaPolyActor_SetPlayerAbove(dynaActor);
     }
 }
 
 void func_80043538(DynaPolyActor* dynaActor) {
-    dynaActor->unk_160 |= 8;
+    dynaActor->interactFlags |= DYNA_INTERACT_3;
 }
 
-s32 func_80043548(DynaPolyActor* dynaActor) {
-    if (dynaActor->unk_160 & 1) {
+s32 DynaPolyActor_HasActorOnTop(DynaPolyActor* dynaActor) {
+    if (dynaActor->interactFlags & DYNA_INTERACT_ACTOR_ON_TOP) {
         return true;
     } else {
         return false;
     }
 }
 
-s32 func_8004356C(DynaPolyActor* dynaActor) {
-    if (dynaActor->unk_160 & 2) {
+s32 DynaPolyActor_HasPlayerOnTop(DynaPolyActor* dynaActor) {
```

Comment:
imo it sounds better without "has"
if (_PlayerOnTop) for example reads fine
I can understand the "has" if youre reading the system name as a part of a sentence, but I personally dont read it that way

Reply:
Imo the "has" (or more commonly "is") in functions like these makes it much clearer that they return a boolean, so it's worth having I think.

Without it the names aren't clear at first glance and potentially confusing (particularly `_ActorOnTop(actor)` which could be interpreted as setting the provided actor as being on top of something else)

Reply:
guess thats fair. i would probably prefer it was Is instead of Has, if we want to have one of them there
IsPlayerOnTop still reads fine